### DISCUSSION TITLE
Zed extension

### DISCUSSION THREAD

**Discussion by hubertmis** [likec4/likec4#1380] [Ideas]
I started using [Zed](https://zed.dev/) as a GUI in my workflow. I like this editor for its performance. 

Was a Likec4 extension for Zed considered to become a part of the Likec4 project? It could be integrated similarly to the VSC extension. Hopefully, with a matching feature set.

**Comment by davydkov** [comment]
Hey @hubertmis 

Since we have LSP, Zed supports LSP, and there are already examples of integration with node-based lsp - I expect no issues in editor.
Also, I expect diagram preview will not work (specific for vscode) and require another implementation.

Do you wanna try to integrate with Zed?

**Reply by davydkov** [reply]
Check [likec4/jetbrains-plugin](https://github.com/likec4/jetbrains-plugin) as an example for custom LSP integration

**Reply by hubertmis** [reply]
Currently, I'm using VSC for likec4 diagrams and Zed for everything else. Lack of live preview is a deal breaker for me. So I'm not going to integrate only the LSP part, and I don't have time or skills to port the preview module

**Comment by almereyda** [comment]
I'm interested in supporting syntax highlighting and LSP usage for a start, and am fine with running a parallel process for testing in a separate browser. Where should I start?

**Reply by davydkov** [reply]
Hey, I would suggest to check #1459 and [likec4/jetbrains-plugin](https://github.com/likec4/jetbrains-plugin)
Then you need to check Zed examples for custom LSP based on nodejs

For syntax highlighting.
As I remember Zed is using treesitter. So we need to make it.
I think we have two options to investigate:
- convert langium grammar [like-c4.langium](https://github.com/likec4/likec4/blob/main/packages/language-server/src/like-c4.langium)
- convert basic TextMate grammar [likec4.tmLanguage.json](https://github.com/likec4/likec4/blob/main/packages/vscode/likec4.tmLanguage.json)

First one looks promising, maybe even better to check first with [langium guys](https://github.com/eclipse-langium/langium/discussions)

On, and there is also semantic syntax highlighting coming from LSP

**Reply by almereyda** [reply]
Thank you. Let me look into this and see what I can come up with.

**Reply by gauthsvenkat** [reply]
Hey @almereyda , did you make any progress? Quite interested in this.

**Comment by Yash-xoxo** [comment]
Hey! Since Zed uses Tree-sitter and there's already LSP support via node-based LSP, here are some quick thoughts:

**For syntax highlighting:**
- Check if Tree-sitter has an Ilkec4 grammar already (or C4/PlantUML as base)
- If not, converting TextMate grammar is probably easier than building from scratch
- langium-guys might have tooling to help

**For LSP integration:**
The jetbrains-plugin example shows how to wire up node-based LSP. You'd need:
1. Package the LSP server
2. Add Zed extension config
3. Wire up language server protocol

**Quick start:**
```json
// extension.json
{
  "name": "ilkec4",
  "language_servers": {
    "ilkec4-lsp": {
      "command": "node",
      "args": ["path/to/lsp/server.js"]
    }
  }
}
```

Semantic highlighting from LSP would be a nice bonus once basic syntax works.

Happy to test if you get a prototype going!
